File: common/src/main/java/org/tdf/common/trie/DumpKeys.java
package org.tdf.common.trie;

import lombok.AccessLevel;
import lombok.Getter;
import org.tdf.common.util.ByteArraySet;
import org.tdf.common.util.HexBytes;

import java.util.HashSet;
import java.util.Set;

@Getter(AccessLevel.PACKAGE)
class DumpKeys implements ScannerAction {
    private Set<HexBytes> keys = new HashSet<>();

    @Override
    public Boolean apply(TrieKey path, Node node) {
        if (node.getHash() != null) {
            keys.add(HexBytes.fromBytes(node.getHash()));
        }
        return true;
    }
}

File: common/src/main/java/org/tdf/common/trie/DumpKeys.kt
package org.tdf.common.trie

import org.tdf.common.util.HexBytes
import org.tdf.common.util.hex

internal class DumpKeys : ScannerAction {
    internal val keys: MutableSet<HexBytes> = HashSet()

    override fun scan(path: TrieKey, node: Node): Boolean {
        if (node.hash != null) {
            keys.add(node.hash.hex())
        }
        return true
    }
}